osu.Game/Graphics/Containers/ScalingContainer.cs

```diff
@@ -111,6 +118,15 @@ private void load(OsuConfigManager osuConfig)
 
             protected override void Update()
             {
+                // If the window size constitutes an aspect ratio smaller than the base size,
+                // then the DrawSizePreservingFillContainer starts descaling the game to ensure it always fits.
+                // We do not desire this behaviour when the window dimensions are small already (e.g. 852x393 on iPhone 16).
+                // The number 576 is picked arbitrarily here as a cutoff point which keeps the game scaled appropriately.
+                float minimumScalingHeight = Math.Min(576f, host.Window.Size.Width / base_aspect_ratio);
```

Comment:
Seems like this is causing a some testing errors, knee jerk reaction is that maybe host.Window.Size is not set when the test calls this method? 🤔

Reply:
it's because there's no actual window when tests are ran headless and `host.Window` is nullrefing. checking this with a debugger takes 5 (five) seconds

Reply:
https://github.com/ppy/osu/pull/31083/commits/9d9b4026081c99365a34e0a794a6d991b32301d4